snuba/clusters/cluster.py

```diff
@@ -316,8 +312,12 @@ def __get_cluster_nodes(self, cluster_name: str) -> Sequence[ClickhouseNode]:
     _unique_registered_storage_sets
 ), "Storage set registered to more than one cluster"
 
+expected_storage_sets = {
+    s for s in StorageSetKey if s.value not in settings.DISABLED_STORAGE_SETS
+}
+
 assert (
-    set(StorageSetKey) == _unique_registered_storage_sets
+    not expected_storage_sets - _unique_registered_storage_sets
```

Comment:
I think disabling a storage set while it is still associated to a cluster makes sense. No need to delete that config to disable it.